manubot/cite/tests/test_cite_command.py

```diff
@@ -65,38 +61,101 @@ def test_cite_command_file(tmpdir):
     not shutil.which('pandoc-citeproc'),
     reason='pandoc-citeproc installation not found on system'
 )
-def test_cite_command_render_stdout(args, expected):
+class Base_cite_command_render_stdout():
     """
-    Test the stdout output of `manubot cite --render` with various formats.
-    The output is sensitive to the version of Pandoc used, so rather than fail when
-    the system's pandoc is outdated, the test is skipped.
+    Expecting reference values for test to be at files on path:
+    cite-command-rendered/references-{format}-{pandoc_stamp}.{extension}
+
+    Examples:
+    - references-html-2.7.2.html  
+    - references-plain-2.7.2.txt
+    - references-jats-2.7.2.xml
+    - references-jats-2.7.3.xml
+    - references-markdown-2.7.2.md
+
+    2.7.2 is current pandoc version on CI builds for these builds.
+
+    Filenames must be adjusted accodingly when current pandoc version changes.
+
+    Run tests locally skipping this test suit (makes sense if your local pandoc
+    version is different from pandoc version used by Travis and Appveyor):
+        pytest -v -m "not pandocfail"
+
+    See .travis.yml and .appveyor.yml to find out current pandoc version used 
+    for testing.
     """
-    pandoc_version = get_pandoc_info()['pandoc version']
-    for output in 'markdown', 'html', 'jats':
-        if output in args and pandoc_version < (2, 5):
-            pytest.skip(f"Test {output} output assumes pandoc >= 2.5")
-    if pandoc_version < (2, 0):
-        pytest.skip("Test requires pandoc >= 2.0 to support --lua-filter and --csl=URL")
-    expected = (
-        pathlib.Path(__file__).parent
-        .joinpath('cite-command-rendered', expected)
-        .read_text()
-    )
-    args = [
-        'manubot', 'cite', '--render',
-        '--csl', 'https://github.com/greenelab/manubot-rootstock/raw/e83e51dcd89256403bb787c3d9a46e4ee8d04a9e/build/assets/style.csl',
-        'arxiv:1806.05726v1', 'doi:10.7717/peerj.338', 'pmid:29618526',
-    ] + args
-    process = subprocess.run(
-        args,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        universal_newlines=True,
-    )
-    print(shlex_join(process.args))
-    print(process.stdout)
-    print(process.stderr)
-    assert process.stdout == expected
+    pandoc_stamp = '.'.join(map(str, pandoc_version))
+    
+    @classmethod
+    def expected_output(cls, format, extension):
+        filename = f'references-{format}-{cls.pandoc_stamp}.{extension}'   
+        return (
+            pathlib.Path(__file__).parent
+            .joinpath('cite-command-rendered', filename)
+            .read_text()
+        )
+        # Note: We wanted to introduce encoding='utf-8-sig' above
+        # for compatability, but that makes tests fail on Travis for 
+        # Windows default encoding.
+
+    @classmethod
+    def render(self, format_args):
```

Comment:
decorator is classmethod, but method defition uses `self` not `cls` and code uses neither, such that it probably should be `@staticmethod`?

Reply:
Indeed! `self` is not used here.